app/services/nsm/all_adjustments_deleter.rb

```diff
@@ -0,0 +1,64 @@
+module Nsm
+  class AllAdjustmentsDeleter < AdjustmentDeleterBase
+    attr_reader :params, :current_user, :submission, :comment
+
+    def initialize(params, adjustment_type, current_user)
+      super
+      @comment = params[:nsm_delete_adjustments_form][:comment]
+    end
+
+    def call
```

Comment:
You might be able to simplify this. In PA we have a rule where when a submission is sent back we want to revert all adjustments, and we simply pull a clean copy of the blob from the app store and replace our adjusted blob with the clean one - https://github.com/ministryofjustice/laa-assess-crime-forms/blob/main/app/forms/prior_authority/send_back_form.rb#L57-L60. (I originally build something to explicitly revert each field like you've done, then David pointed out this as a simpler approach).

Reply:
Cheers! one thing though wouldn't nuking all the adjustments like that obviate the item event tracking we get if we iterate through reverts? do we even need to track those events (we're doing it in PA, so i'm assuming that we need to do that for _some reason_)

Reply:
So if we did it that way we'd end up with a series of Edit events (the event history is outside the blob so wouldn't get nuked) and a single DeleteAdjustments event. We wouldn't have a set of individual UndoEdit events, but I kinda think that's reasonable - because there hasn't _been_ a series of undo events in the colloquial sense, there's just been one event which is the provider clicking the 'Delete all adjustments' button.

TBH what we use Event objects for has changed a lot. We used to treat them as things that could track very precisely the changes in state of an application over time, but these days we really only use them for the event history UI and occasional manual debugging/auditing, that's mostly replicated by the Kibana logs etc. I feel like the Edit/UndoEdit events aren't all that useful at this point so perhaps I'm more not-fussed about them than most.

Reply:
ok, i was wondering if we were using those events indirectly to refresh what we had in app-store or other shenanigans but i guess not - i'll make that suggested change as it reduces a bit of clunk